Support loading `stubs/` from typeshed

When I create a file `foo.py` with `import toml` and execute `mypy foo.py --custom-typeshed-dir my-typeshed/` I get:

```
foo.py:1: error: Library stubs not installed for "toml" (or incompatible with Python 3.9)
foo.py:1: note: Hint: "python3 -m pip install types-toml"
foo.py:1: note: (or run "mypy --install-types" to install all missing stub packages)
foo.py:1: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
Found 1 error in 1 file (checked 1 source file)
```

Although `my-typeshed/stubs/` contains `toml/`.

I find this really inconvenient because I'd like to test my third-party stub typeshed contributions with mypy.

It's worth noting that pyright handles this, `pyright foo.py --typeshed-path my-typeshed/` works as expected.

--- a/docs/source/command_line.rst
+++ b/docs/source/command_line.rst
@@ -830,7 +830,8 @@ in developing or debugging mypy internals.
     submitting them upstream, but also allows you to use a forked version of
     typeshed.
 
-    Note that this doesn't affect third-party library stubs.
+    Note that this doesn't affect third-party library stubs. To test third-party stubs,
+    for example try ``MYPYPATH=stubs/six mypy ...``.
 
 .. _warn-incomplete-stub:
 

--- a/docs/source/config_file.rst
+++ b/docs/source/config_file.rst
@@ -858,9 +858,16 @@ These options may only be set in the global section (``[mypy]``).
 
     :type: string
 
-    Specifies an alternative directory to look for stubs instead of the
-    default ``typeshed`` directory. User home directory and environment
-    variables will be expanded.
+    This specifies the directory where mypy looks for standard library typeshed
+    stubs, instead of the typeshed that ships with mypy.  This is
+    primarily intended to make it easier to test typeshed changes before
+    submitting them upstream, but also allows you to use a forked version of
+    typeshed.
+
+    User home directory and environment variables will be expanded.
+
+    Note that this doesn't affect third-party library stubs. To test third-party stubs,
+    for example try ``MYPYPATH=stubs/six mypy ...``.
 
 .. confval:: warn_incomplete_stub
 

